frontend/src/components/headers/DocumentHeader.tsx

```diff
@@ -11,24 +11,51 @@ import {
 	ToggleButtonGroup,
 	Toolbar,
 	Tooltip,
+	Button,
+	FormControl,
 } from "@mui/material";
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
 import { useDispatch, useSelector } from "react-redux";
-import { useNavigate } from "react-router-dom";
+import { useNavigate, useParams } from "react-router-dom";
 import { useUserPresence } from "../../hooks/useUserPresence";
 import { EditorModeType, selectEditor, setMode } from "../../store/editorSlice";
 import { selectWorkspace } from "../../store/workspaceSlice";
 import DownloadMenu from "../common/DownloadMenu";
 import ShareButton from "../common/ShareButton";
 import ThemeButton from "../common/ThemeButton";
 import UserPresenceList from "./UserPresenceList";
+import { FormContainer, TextFieldElement } from "react-hook-form-mui";
+import { useGetWorkspaceQuery } from "../../hooks/api/workspace";
+
+import {
+	useUpdateDocumentTitleMutation,
+	useGetDocumentQuery,
+} from "../../hooks/api/workspaceDocument";
 
 function DocumentHeader() {
 	const dispatch = useDispatch();
 	const navigate = useNavigate();
 	const editorState = useSelector(selectEditor);
 	const workspaceState = useSelector(selectWorkspace);
 	const { presenceList } = useUserPresence(editorState.doc);
+	const [focused, setFocused] = useState(false);
+	const [documentTitle, setDocumentTitle] = useState("");
```

Comment:
How about using `useForm`? With `useForm`, we can handle validation.

Reply:
Is this something i can do Form from MUI or should i:

import { useForm } from "react-hook-form";

Reply:
Sorry, you can use `react-hook-form-mui`. 
Please refer to `frontend/src/components/modals/ChangeNicknameModal.tsx`.